packages/core/src/virtual-module/virtual-routes.ts

```diff
@@ -80,17 +80,26 @@ export function createVirtualRouteComponent(route: VirtualRoute): string {
   return `
 ---
 import StoryPage from 'astrobook/pages/story.astro'
-import { isAstroStory } from 'astrobook/client'
+import { isAstroStory, isAstroComponentFactory } from 'astrobook/client'
 import * as m from '${route.storyModule.importPath}'
 
 const isAstro = isAstroStory(m)
 ---
 
 <StoryPage story={'${route.props.story}'} hasSidebar={${route.props.hasSidebar}}>
   {
-    isAstro
+    (({...m['${route.story.name}']}?.decorators || []).slice().reverse().reduce((currentTree, decoratorFn) => {
+      const Decorator = decoratorFn()
```

Comment:
Is there any reason that `decoratorFn` must be a function? 

In another word, what if we defined the decorator like this in the `.stories.ts` file? 

```diff
   export default {
     component: Button,
     args: { variant: 'secondary' } satisfies ButtonProps,
-    decorators: [() => DarkModeContainer],
+    decorators: [DarkModeContainer],
   }
```